### ISSUE TITLE
Feature properties > color assignment

### ISSUE THREAD

**Issue by gmcarthuriv** [TeselaGen/openVectorEditor#210]
Request:
The ability to assign colors to GenBank features

**Comment by tnrich** [comment]
@gmcarthuriv do you have a preference for what color palette to make accessible? Should any color be available? 

cc @mfero @njhillson

**Comment by njhillson** [comment]
Hi @tnrich there is a standard/branded palette for JBEI. I imagine each
company may want their own custom palette. However, I imagine for any given
instance (company) having a consistent palette/coloring scheme for a given
feature type would be a good idea. Multiple allowed colors for the same
feature type could be confusing...

**Comment by tnrich** [comment]
@njhillson I definitely agree that having a standard color for each feature type is desirable but it sounds to me like @gmcarthuriv wants to be able to sometimes change a given feature's color to a color different than it's default type-based color. Correct me if I'm wrong @gmcarthuriv

**Comment by njhillson** [comment]
Ok, would be good for @gmcarthuriv <https://github.com/gmcarthuriv> to
specify if this is, for example, just a request for an Arzeda-standard
color palette that is distinct from TeselaGen default (but that normal
users can't change), or if he wants any one (pending user role I suppose)
to be able to chose a custom color for any particular feature within the
Arzeda instance.

**Comment by gmcarthuriv** [comment]
@tnrich, @njhillson,

Apologies for the delay. My thoughts:

- "Multiple colors for the same feature type could be confusing..." For common features such as CDSs, yes. However, it would be really handy to be able to color different misc_feature labels with different colors. That is, there are often a myriad of misc_feature labels (as shown in the example below). Color could be a nice parameter to differentiate various sub-types (e.g., insulators).
- Currently there is a lot of color sharing between feature types. For example (again, shown below), misc_feature labels and RBS labels use the same color (grey). In fact, most features use grey (if I remember correctly), which is not very useful.
- It would be really cool to able to assign any color to any feature type using hex codes or by picking from a more-constrained color "theme" specific to Arzeda, but not necessary. That is, being able to pick from a standard set of colors is sufficient (e.g., http://clrs.cc/).

Given the above comments, my suggestion is to allow the user to 1) assign colors (from a standard set) to feature types and 2) allow the creation of multiple misc_feature types to align better with current synthetic biology practices. For example, within the app a user could define a GenBank feature type "insulator" (misc_feature_insulator) to correspond with the SBOL genetic part concept. It would be displayed as a GenBank feature within OpenVE. Upon export, these custom features would be collapsed into a single GenBank feature type "misc_feature" so that it would render correctly in ApE, etc.


![image](https://user-images.githubusercontent.com/17152701/35244043-ac150ae6-ff73-11e7-8e64-7ff22c4d978c.png)

**Comment by tnrich** [comment]
Hey @gmcarthuriv I think what you're saying makes sense. 

You're also totally right about many of the features defaulting to grey. I will set better defaults for those. 

Maybe as a first step I'll make a color picker that lets you choose any color for any feature, and then as a second step allow you to change the default color for given feature types. 

@njhillson the color picker will not be enabled by default, only if a config option is passed that turns it on.

**Comment by njhillson** [comment]
misc_feature subtypes (extending GenBank) sounds fine to me. Upon export
reverting to misc_feature seems sensible.

better standard colors would be good (enabling admin to set these).

I would suggest keeping this an admin feature, to set the color for each
feature type for the whole instance. However, if @gmcarthuriv / Arzeda
wants each user to be able to independently to do this, one question would
be do I look at someone else's plasmid map with my preferred pallet, or the
one they used? Just a thought experiment...

**Comment by gmcarthuriv** [comment]
I agree; it would be best to implement this across the entire company (i.e., an admin determines the color scheme and is locked to all other users).

**Comment by gmcarthuriv** [comment]
@tnrich, @tiffanydai, @craftjk, @siobhandavis,

Is this ticket still active? Please let me know if there's anything I can do to help.
